internal/module/statusupdater.go

```diff
@@ -53,5 +57,42 @@ func (s *statusUpdater) UpdateModuleStatus(ctx context.Context, mod *ootov1alpha
 		mod.Status.DevicePlugin.DesiredNumber = numDesired
 		mod.Status.DevicePlugin.AvailableNumber = numAvailableDevicePlugin
 	}
-	return s.client.Status().Patch(ctx, mod, client.MergeFrom(original))
+	s.updateMetrics(ctx, mod, targetedNodes, kernelMappingNodes)
+	return s.client.Status().Update(ctx, mod)
+}
+
+func (s *statusUpdater) updateMetrics(ctx context.Context, mod *ootov1alpha1.Module, targetedNodes, kernelMappingNodes []v1.Node) {
+	logger := log.FromContext(ctx)
+	s.metricsAPI.SetCompletedStage(mod.Name, mod.Namespace, metrics.DriverContainerStage, mod.Status.DriverContainer.DesiredNumber == mod.Status.DriverContainer.AvailableNumber)
+	if mod.Spec.DevicePlugin != nil {
+		s.metricsAPI.SetCompletedStage(mod.Name, mod.Namespace, metrics.DevicePluginStage, mod.Status.DevicePlugin.DesiredNumber == mod.Status.DevicePlugin.AvailableNumber)
+	}
+
+	deployedNodes, err := s.getCurrentlyDeployedNodes(ctx, mod)
+	if err != nil {
+		logger.Info("failed to get deployed nodes", "module", mod.Name, "error", err)
```

Comment:
How about returning that error instead?

Reply:
not relevant anymore, as we won't supply node utilization metrics